cmd/vcluster/cmd/hostpaths.go

```diff
@@ -0,0 +1,473 @@
+	wait.Forever(func() {
+		podList := &corev1.PodList{}
+		err := pManager.GetClient().List(ctx, podList, &client.ListOptions{Namespace: options.TargetNamespace})
+		if err != nil {
+			klog.Errorf("unable to list pods: %v", err)
+			return
+		}
+
+		podMappings := make(PhysicalPodMap)
+
+		fillUpPodMapping(ctx, podList, podMappings)
+
+		// TODO: remove these log lines
+		// klog.Infof("podMapping length: %d", len(podMappings))
+		// klog.Infof("podList length: %d", len(podList.Items))
+
+		vPodList := &corev1.PodList{}
+		err = vManager.GetClient().List(ctx, vPodList)
+		if err != nil {
+			klog.Errorf("unable to list pods: %v", err)
+			return
+		}
+
+		existingVPodsWithNamespace := make(map[string]bool)
+		existingPodsPath := make(map[string]bool)
+		existingKubeletPodsPath := make(map[string]bool)
+
+		for _, vPod := range vPodList.Items {
+			existingVPodsWithNamespace[fmt.Sprintf("%s_%s", vPod.Name, vPod.Namespace)] = true
+			pName := translate.PhysicalName(vPod.Name, vPod.Namespace)
+
+			if podDetail, ok := podMappings[pName]; ok {
+				// create pod log symlink
+				source := filepath.Join(virtualPodLogsPath, fmt.Sprintf("%s_%s_%s", vPod.Namespace, vPod.Name, string(vPod.UID)))
+				target := filepath.Join(pods.PhysicalLogVolumeMountPath, podDetail.Target)
+
+				existingPodsPath[source] = true
+
+				_, err := createPodLogSymlinkToPhysical(ctx, source, target)
+				if err != nil {
+					klog.Errorf("unable to create symlink for %s: %v", podDetail.Target, err)
+				}
+
+				// create kubelet pod symlink
+				kubeletPodSymlinkSource := filepath.Join(virtualKubeletPodPath, string(vPod.GetUID()))
+				kubeletPodSymlinkTarget := filepath.Join(pods.PhysicalKubeletVolumeMountPath, string(podDetail.PhysicalPod.GetUID()))
+				existingKubeletPodsPath[kubeletPodSymlinkSource] = true
+				createKubeletVirtualToPhysicalPodLinks(ctx, kubeletPodSymlinkSource, kubeletPodSymlinkTarget)
+
+				// podDetail.SymLinkName = symlinkName
+
+				// create container to vPod symlinks
+				containerSymlinkTargetDir := filepath.Join(PodLogsMountPath,
+					fmt.Sprintf("%s_%s_%s", vPod.Namespace, vPod.Name, string(vPod.UID)))
+				createContainerToPodSymlink(ctx, vPod, podDetail, containerSymlinkTargetDir)
+			}
+		}
+
+		// cleanup old pod symlinks
+		err = cleanupOldPodPath(ctx, virtualPodLogsPath, existingPodsPath)
+		if err != nil {
+			klog.Errorf("error cleaning up old pod log paths: %v", err)
+		}
+
+		// cleanup old container symlinks
+		// QUESTION: current mechanism cleans up container symlinks
+		// for old pods, but does not cleanup for container restarts
+		// should we be supporting that too?
+		err = cleanupOldContainerPaths(ctx, existingVPodsWithNamespace)
+		if err != nil {
+			klog.Errorf("error cleaning up old container log paths: %v", err)
+		}
+
+		// QUESTION: should we clear this immediately, would there be
+		// repercussions as we might miss some backups if we delete immediately
```

Comment:
@FabianKramm I would like to know your thoughts about these 2 scenarios

Reply:
How is this handled in a regular cluster? Are those cleared immediately? In general the goal is here to stay as close as possible to the behaviour of a physical cluster, so what ever the physical cluster is doing here, we should try to replicate that behvaiour

Reply:
we are now checking if the internal symlinks resolve. We only delete once they stop resolving